### ISSUE TITLE
How to update texts

### ISSUE THREAD

**Issue by itmuckel** [ggez/ggez#281]
Hey guys! 
I've finally updated my project to ggez 0.4 and I love the progress. This is my favorite rust project. :smile:

Now that I got the basic structure of my game, I want to update some texts on specific events or continuously (for example: slowly decreasing HP of a player's character). 
Currently I'm doing this by creating new instances of `ggez::graphics::Text` every time the value changes.

This works, but it feels wrong. Is there a way to update the text's contents without recreating (reallocating) a new text instance? If no, is this performance-wise not critical e.g. the way to go here? How are you guys doing it?

Greetings,
itmuckel

**Comment by icefoxen** [comment]
Nope, recreating the Text is the best way currently. It is not a fast
operation, so you don't want to do it every frame if the text doesn't
change, but in my experience the cost of memory allocation is far smaller
than the cost of all the math needed to render the font.

The obvious optimization is to assemble text objects out of pre rendered
chunks, which we want to get to eventually but we just aren't there yet.
See issues #132 and #270 for discussion on that.


Feb 11, 2018 03:58, "itmuckel" <notifications@github.com> wrote:

Hey guys!
I've finally updated my project to ggez 0.4 and I love the progress. This
is my favorite rust project. 😄

Now that I got the basic structure of my game, I want to update some texts
on specific events or continuously (for example: slowly decreasing HP of a
player's character).
Currently I'm doing this by creating new instances of ggez::graphics::Text
every time the value changes.

This works, but it feels wrong. Is there a way to update the text's
contents without recreating (reallocating) a new text instance? If no, is
this performance-wise not critical e.g. the way to go here? How are you
guys doing it?

Greetings,
itmuckel

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/ggez/ggez/issues/281>, or mute the thread
